src/app/editor-common/components/key-value-component/key-value-input.component.ts

```diff
@@ -0,0 +1,129 @@
+import {Component, forwardRef, Input} from "@angular/core";
+import {
+    NG_VALUE_ACCESSOR,
+    ControlValueAccessor,
+    FormGroup,
+    FormBuilder,
+    NG_VALIDATORS,
+    FormControl
+} from "@angular/forms";
+import {ExpressionModel} from "cwlts/models/d2sb";
+import {ComponentBase} from "../../../components/common/component-base";
+
+require("./key-value-input.component.scss");
+
+@Component({
+    selector: "[ct-key-value-input]",
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => KeyValueInputComponent),
+            multi: true
+        },
+        {
+            provide: NG_VALIDATORS,
+            useExisting: forwardRef(() => KeyValueInputComponent),
+            multi: true
+        }
+    ],
+    template: `
+
+            <div class="key-container col-sm-5 input-group">
+                 <i class="col-sm-1 fa fa-times-circle validation-icon key-validation-icon"
+                    [ct-tooltip]="keyTooltip"
+                    *ngIf="!form.controls['keyForm'].valid"></i>
+                
+                <ct-tooltip-content #keyTooltip>
+                     <div class="error-text px-1" *ngFor="let error of errorMessages">{{ error }}</div>
+                </ct-tooltip-content>
+            
+                <input class="ellipsis form-control key-input"
+                   [class.col-sm-12]="form.controls['keyForm'].valid"
+                   [class.col-sm-11]="!form.controls['keyForm'].valid"
+                   *ngIf="form.controls['keyForm']"
+                   [formControl]="form.controls['keyForm']"/>
+            </div>
+               
+            <ct-expression-input 
```

Comment:
If this component is to be used on the hints section, it needs to not expect all values to be instances of `ExpressionModel`. Some hints can have arbitrary object literal values, others can have string literal values and so on.